autoconnect/autoconnect-web/src/routes.rs

```diff
@@ -43,9 +43,11 @@ pub async fn push_route(
             .await;
     }
     // Attempt to send the notification to the UA using WebSocket protocol, or store on failure.
+    // NOTE: Since this clones the notification, there is a potential to
+    // double count the reliability state.
     let result = app_state
         .clients
-        .notify(uaid.into_inner(), notif.clone())
+        .notify(uaid.into_inner(), notif.clone_without_reliability_state())
```

Comment:
what state are you seeing double counted? nulling out the state here seems like it would guard against the `old` state being double decremented

Reply:
I'm seeing some older states like `Received` which are being double counted. My theory is that this happens because of a hand-off of a cloned item. (The original and cloned item both would have the old state `Received`, which would be decremented by the new states.)

Reply:
I _think_ there was an issue with double counting `Received` because of a hand-off through internal routing, but I agree, that's only a theory. (This is also one of the reasons that I avoid state machines whenever possible. I've had a lot of weird, subtle bugs that crop up because of some strange relationship or impact of using them. Yes, it's a dumb fear, but it's my dumb fear.)